### PR TITLE
[JS] Topic: Loops

### LINKED ISSUE #380: JS: Loops
## Directory

`Topics/5.Syntax/Loops`

The index will be added manually after merging.

## Description

- What are loops?
- Types of loops
- Syntax of each loop
- Code Snippets of each type of loop

## Formats

- [x]  Document
- [ ]  Video
- [ ] Audio

### PR DESCRIPTION
**Pull request by harshmathurx**
<hr>

## Description 📜

Added folders for each type of loop
Added content in each folder with .md files 
Fixes #380

<hr>

## Type of change 📝
- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [ ] Android Dev (Flutter, Java, Kotlin)
- [ ] Backend Dev (Java, .NET, Python)
- [ ] C/CPP
- [ ] Competitive Programming
- [ ] Cyber Security
- [ ] DSA
- [ ] Database
- [ ] Datascience with Python
- [ ] Datascience with R
- [ ] Frontend Dev (HTML, CSS, JS)
- [ ] Frontend Dev (React, Angular, Vue)
- [ ] Golang
- [ ] Interview Prep
- [ ] Java
- [x] JavaScript
- [ ] MERN
- [ ] Machine Learning
- [ ] Open Source
- [ ] Python
- [ ] Rust
- [ ] Statistics
- [ ] UI/UX

<hr>
 
## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm a GWOC'21 contributor

<hr>

### CODE REVIEW

**Comment by harshmathurx** [comment]
@paramsiddharth can you please review this PR

**Comment by harshmathurx** [comment]
This solves #380

**Comment by dakshsethi** [comment]
Hi @harshmathurx this is great indeed. First of all rename the main **Loops.md** file to **READEME.md**.
Also try to include work-flow diagram of each and every type of loop.
Also we can include ForEach loop as well in this.

**Comment by harshmathurx** [comment]
all right I'll make those changes :)

**Comment by harshmathurx** [comment]
@dakshsethi hey, I made the changes as you suggested

**Review by paramsiddharth — APPROVED** [review]
Really loved the modularization! I appreciate such practices.

@Creatoon @dakshsethi Please review. It looks good to me.

File: Web_Development/Javascript/Topics/5.Syntax/Loops/For_Loop/For_Loop.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- **Initialization condition:** Here, we initialize the
+variable in use. It marks the start of a for loop. An already declared
+variable can be used or a variable can be declared, local to loop only.
+- **Testing Condition:** It is used for testing the exit condition for a loop. It must return a boolean value. It is also an **Entry Control Loop** as the condition is checked prior to the execution of the loop statements.
+- **Statement execution:** Once the condition is evaluated to true, the statements in the loop body are executed.
+- **Increment/ Decrement:** It is used for updating the variable for next iteration.
+- **Loop termination:** When the condition becomes false, the loop terminates marking the end of its life cycle.
+
+![](For_loop_1.png)
+
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/For_Loop/For_Loop.md:22]: Photo is not visible!!

File: Web_Development/Javascript/Topics/5.Syntax/Loops/For_Each_Method/For_Each.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+array.forEach(*function(currentValue, index, arr), thisValue*)
+```
+
+The `forEach` method passes a callback function for each element of an array together with the following parameters:
+
+- Current Value (required) - The value of the current array element
+- Index (optional) - The current element's index number
+- Array (optional) - The array object to which the current element belongs
+
+![Untitled](For_each_1.png)
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/For_Each_Method/For_Each.md:19]: This image is not showing!!

File: Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/Do_While.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+1. do while loop starts with the execution of the statement(s). There is no checking of any condition for the first time.
+2. After the execution of the statements, and update of the variable value, the
+condition is checked for true or false value. If it is evaluated to
+true, next iteration of loop starts.
+3. When the condition becomes false, the loop terminates which marks the end of its life cycle.
+4. It is important to note that the do-while loop will execute its statements atleast once before any condition is checked, and therefore is an
+example of exit control loop.
+
+![](Do_while_1.png)
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/Do_While.md:24]: Image not showing!!

**Comment by harshmathurx** [comment]
thanks. shall I close this issue now?

File: Web_Development/Javascript/Topics/5.Syntax/Loops/While_Loop/While_Loop.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+```
+while (boolean condition)
+{
+   loop statements...
+}
+
+```
+
+Control Flow of a while loop
+![](While_loop_1.png)
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/While_Loop/While_Loop.md:16]: Image not showing!!

File: Web_Development/Javascript/Topics/5.Syntax/Loops/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+end of loop body. Therefore, the loop body will execute atleast once,
+irrespective of whether the test condition is true or false. **do – while loop** is exit controlled loop.   
+
+The different types of statements to use loops are in JavaScript are :
+
+- For loop
+- For in loop
+- While loop
+- do-while loop
+
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/README.md:50]: Add For Each loop here also.

**Comment by dakshsethi** [comment]
> thanks. shall I close this issue now?

No wait, there are still some issues.

File: Web_Development/Javascript/Topics/5.Syntax/Loops/For_Loop/For_Loop.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# For Loop
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/For_Loop/For_Loop.md:1]: Rename this file to `README.md`

File: Web_Development/Javascript/Topics/5.Syntax/Loops/While_Loop/While_Loop.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# While Loop
+
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/While_Loop/While_Loop.md:2]: Rename the file to `README.md`

File: Web_Development/Javascript/Topics/5.Syntax/Loops/For_In_Loop/For_In.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+# For in loop
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/For_In_Loop/For_In.md:1]: Rename the file to `README.md`

File: Web_Development/Javascript/Topics/5.Syntax/Loops/For_Each_Method/For_Each.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
+# For Each Method
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/For_Each_Method/For_Each.md:1]: Rename the file to `README.md`

File: Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/Do_While.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
+# Do While Loop
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/Do_While.md:1]: Rename the file to `README.md`

**Comment by Creatoon** [comment]
@harshmathurx we can not continue or break from forEach loop, so there is another kind of loop `every`, Read about it once and mention this also.

Screenshot_20210918-102155

**Review by dakshsethi — CHANGES_REQUESTED** [review]

File: Web_Development/Javascript/Topics/5.Syntax/Loops/For_Loop/For_Loop.md (hunk 1 above)
Comment (harshmathurx) [reply Web_Development/Javascript/Topics/5.Syntax/Loops/For_Loop/For_Loop.md:22]: It is visible. I don't think you can see it in this PR message
check this out: https://github.com/harshmathurx/winter-of-contributing/blob/Javascript/Web_Development/Javascript/Topics/5.Syntax/Loops/For_Loop/For_Loop.md

**Comment by harshmathurx** [comment]
@dakshsethi @Creatoon @paramsiddharth I've made the changes

**Comment by paramsiddharth** [comment]
@harshmathurx @dakshsethi Please review.

File: Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
+# Do While Loop
+
+do while loop is similar to while loop with only difference that it checks for condition after executing the statements, and therefore is an example of Exit Control Loop.
```
Comment (Creatoon) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md:3]:
```suggestion
do while loop is similar to while loop with only difference that it checks for the condition after executing the statements, and therefore it is an example of Exit Control Loop.
```

File: Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+while (condition);
+
+```
+
+1. do while loop starts with the execution of the statement(s). There is no checking of any condition for the first time.
+2. After the execution of the statements, and update of the variable value, the
+condition is checked for true or false value. If it is evaluated to
+true, next iteration of loop starts.
+3. When the condition becomes false, the loop terminates which marks the end of its life cycle.
+4. It is important to note that the do-while loop will execute its statements atleast once before any condition is checked, and therefore is an
```
Comment (Creatoon) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md:21]:
```suggestion
4. It is important to note that the do-while loop will execute its statements atleast once before any condition is checked, and therefore it is an
```

File: Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+loop tests the condition. So, Do While executes the statements in the
+code block at least once even if the condition Fails.
+
+### Example
+
+```jsx
+		var x = 21;
+    do 
+    {
+        // The line while be printer even
```
Comment (Creatoon) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md:42]: I don't printer is right word, change if wrong

**Review by Creatoon — CHANGES_REQUESTED** [review]

File: Web_Development/Javascript/Topics/5.Syntax/Loops/For_Loop/For_Loop.md (hunk 1 above)
Comment (paramsiddharth) [reply Web_Development/Javascript/Topics/5.Syntax/Loops/For_Loop/For_Loop.md]: Yes @dakshsethi , the web-diff sometimes doesn't render the images. :)

File: Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md (hunk 13 above)
Comment (paramsiddharth) [reply Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md:42]: @harshmathurx *Printed. It is a typo.

**Comment by paramsiddharth** [comment]
@harshmathurx Look for the abovementioned typo and fix it. Once done, your PR will be ready to merge.

**Comment by harshmathurx** [comment]
@paramsiddharth fixed the typo, thank you @Creatoon for mentioning it

**Review by Creatoon — APPROVED** [review]

**Comment by s4-2** [comment]
Hii @harshmathurx , Nice job. The format of writing is really nice and simple to understand. I request you to please in description add **Fixes #380** as it will provide any contributor or mentor to see which issue this PR will solve.
Hope you get it, feel free to ask any query:-)

File: Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+1. do while loop starts with the execution of the statement(s). There is no checking of any condition for the first time.
+2. After the execution of the statements, and update of the variable value, the
+condition is checked for true or false value. If it is evaluated to
+true, next iteration of loop starts.
+3. When the condition becomes false, the loop terminates which marks the end of its life cycle.
+4. It is important to note that the do-while loop will execute its statements atleast once before any condition is checked, and therefore it is an
+example of exit control loop.
+
+![](Do_while_1.png)
+
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/5.Syntax/Loops/Do_While_Loop/README.md:25]: Provide `alt text` to all of your images.

**Review by dakshsethi — CHANGES_REQUESTED** [review]

**Comment by harshmathurx** [comment]
> 
> 
> Hii @harshmathurx , Nice job. The format of writing is really nice and simple to understand. I request you to please in description add **Fixes #380** as it will provide any contributor or mentor to see which issue this PR will solve.
> Hope you get it, feel free to ask any query:-)

Hey can you guide me how to do that?

**Review by dakshsethi — APPROVED** [review]
Great work @harshmathurx !!

**Comment by harshmathurx** [comment]
> Hii @harshmathurx , Nice job. The format of writing is really nice and simple to understand. I request you to please in description add **Fixes #380** as it will provide any contributor or mentor to see which issue this PR will solve.
> Hope you get it, feel free to ask any query:-)

done, thank you :)

**Pull request merged**